src/app/components/Register/RegisterForm.tsx

```diff
@@ -0,0 +1,123 @@
+import { useState } from 'react';
+import TextField from '@mui/material/TextField';
+import AccountCircleIcon from '@mui/icons-material/AccountCircle';
+import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
+import { Typography } from '@mui/material';
+import { register } from '../../../../lib/services/registerService';
+import Alert from '@mui/material/Alert';
+
+const styles = {
+  header: {
+    display: 'flex',
+    justifyContent: 'center',
+  },
+  loginIconStyles: (theme: any) => ({
+    color: `${theme.palette.WHITE}`,
+    paddingLeft: '20px',
+  }),
+  formBox: (theme: any) => ({
+    backgroundColor: `${theme.palette.BEIGE}`,
+    position: 'absolute',
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)',
+    padding: '30px',
+    borderRadius: '12px',
+  }),
+  registerButton: {
+    margin: '20px 0',
+  },
+};
+
+const RegisterForm: React.FC = () => {
+  const [email, setEmail] = useState('');
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [error, setError] = useState('');
+  const [success, setSuccess] = useState(false);
+
+  const handleRegister = async (e: React.FormEvent<HTMLFormElement>) => {
```

Comment:
generalnie spoko, ale przemyślałbym użycie jednak biblioteki do tego. Im szybciej spróbujesz używania Formik/ React Forms / cokolwiek innego - tym lepiej. pisanie samemu to koło od nowa - a w większości miejsc raczej spotkasz się z użyciem biblioteki